simulator/simulator.go

```diff
@@ -0,0 +1,124 @@
+package simulator
+
+import (
+	"errors"
+	"sync"
+)
+
+// FakeHost is the interface which defines the pluggable behavior of a simulated
+// host. Each "type" of host may have their own implementation of this interface.
+type FakeHost interface {
```

Comment:
I think we should probably rename this interface to `Osquery` or `FakeOsquery`.

Reply:
Why not QueryRunner?